### PR TITLE
Cldc 809 supported housing questions

### PR DESCRIPTION
**Pull request by kosiakkatrina**
Add a unittype_sh column (equivalent to unittype_gn for shared housing)
Add display subsection field to the json (we don't want to display the subsection for supported housing renewals as we can infer the values for those)
Display the correct amount of completed sections
Update the subsection status to only return not started if there are at least 1 applicable questions

### CODE REVIEW

File: app/models/derived_variables/case_log_variables.rb [hunk 1]
```diff
@@ -67,8 +68,29 @@ def set_derived_fields!
...
+        type_of_unit_map = {
+          "Self-contained flat or bedsit" => 1,
+          "Self-contained flat or bedsit with common facilities" => 2,
+          "Shared flat" => 3,
+          "Shared house or hostel" => 4,
+          "Bungalow" => 5,
+          "Self-contained house" => 6,
+        }
+        self.unittype_sh = type_of_unit_map[location.type_of_unit]
+        self.builtype = form.questions.find { |x| x.id == "builtype" }.answer_options.select { |_key, value| value["value"] == location.type_of_building }.keys.first
```
Comment (baarkerlounger) [inline app/models/derived_variables/case_log_variables.rb]: Just syntactic sugar but we have `form.get_question("builtype", self)`, probably wants to be a `.find` on the answer options so you stop looping as soon as you have a match.

File: app/models/derived_variables/case_log_variables.rb [hunk 2]
```diff
@@ -67,8 +68,29 @@ def set_derived_fields!
...
-    if is_supported_housing? && (scheme && scheme.locations.size == 1)
-      self.location = scheme.locations.first
+    if is_supported_housing? && scheme
+      if scheme.locations.size == 1
+        self.location = scheme.locations.first
+      end
+      if location
+        self.la = location.county
+        self.postcode_full = location.postcode
+        type_of_unit_map = {
```
Comment (baarkerlounger) [inline app/models/derived_variables/case_log_variables.rb]: Worth extracting this into a constant?

**Comment by baarkerlounger** [comment]
Looks like there's still a looping query:

```ruby
Started POST "/logs/2026/form" for 127.0.0.1 at 2022-07-11 09:36:29 +0100
Processing by FormController#submit_form as HTML
  Parameters: {"authenticity_token"=>"[FILTERED]", "case_log"=>{"rent_type"=>"1", "irproduct_other"=>"", "page"=>"rent_type"}, "id"=>"2026"}
  User Load (0.8ms)  SELECT "users".* FROM "users" WHERE "users"."id" = $1 ORDER BY "users"."id" ASC LIMIT $2  [["id", 2], ["LIMIT", 1]]
  Organisation Load (0.7ms)  SELECT "organisations".* FROM "organisations" WHERE "organisations"."id" = $1 LIMIT $2  [["id", 1], ["LIMIT", 1]]
  ↳ app/models/user.rb:44:in `case_logs'
  CaseLog Load (0.9ms)  SELECT "case_logs".* FROM "case_logs" WHERE ("case_logs"."owning_organisation_id" = $1 OR "case_logs"."managing_organisation_id" = $2) AND "case_logs"."id" = $3 LIMIT $4  [["owning_organisation_id", 1], ["managing_organisation_id", 1], ["id", 2026], ["LIMIT", 1]]
  ↳ app/controllers/form_controller.rb:96:in `find_resource'
  TRANSACTION (1.3ms)  BEGIN
  ↳ app/models/case_log.rb:518:in `reset_created_by'
  User Load (0.9ms)  SELECT "users".* FROM "users" WHERE "users"."id" = $1 LIMIT $2  [["id", 2], ["LIMIT", 1]]
  ↳ app/models/case_log.rb:518:in `reset_created_by'
  CACHE Organisation Load (0.0ms)  SELECT "organisations".* FROM "organisations" WHERE "organisations"."id" = $1 LIMIT $2  [["id", 1], ["LIMIT", 1]]
  ↳ app/models/case_log.rb:518:in `reset_created_by'
  CACHE Organisation Load (0.0ms)  SELECT "organisations".* FROM "organisations" WHERE "organisations"."id" = $1 LIMIT $2  [["id", 1], ["LIMIT", 1]]
  ↳ app/models/case_log.rb:520:in `reset_created_by'
  Scheme Load (1.5ms)  SELECT "schemes".* FROM "schemes" WHERE "schemes"."id" = $1 LIMIT $2  [["id", 1], ["LIMIT", 1]]
  ↳ app/models/derived_variables/case_log_variables.rb:9:in `scheme_has_multiple_locations?'
  Location Count (1.4ms)  SELECT COUNT(*) FROM "locations" WHERE "locations"."scheme_id" = $1  [["scheme_id", 1]]
  ↳ app/models/derived_variables/case_log_variables.rb:11:in `scheme_has_multiple_locations?'
  CACHE Location Count (0.0ms)  SELECT COUNT(*) FROM "locations" WHERE "locations"."scheme_id" = $1  [["scheme_id", 1]]
  ↳ app/models/derived_variables/case_log_variables.rb:11:in `scheme_has_multiple_locations?'
  CACHE Location Count (0.1ms)  SELECT COUNT(*) FROM "locations" WHERE "locations"."scheme_id" = $1  [["scheme_id", 1]]
  ↳ app/models/derived_variables/case_log_variables.rb:11:in `scheme_has_multiple_locations?'
  CACHE Location Count (0.0ms)  SELECT COUNT(*) FROM "locations" WHERE "locations"."scheme_id" = $1  [["scheme_id", 1]]

```

File: app/models/derived_variables/case_log_variables.rb (hunk 1 above)
Comment (kosiakkatrina) [reply app/models/derived_variables/case_log_variables.rb]: select doesn't fail if it doesn't find anything (that shouldn't be the case, but it is in our tests)

Comment (baarkerlounger) [reply app/models/derived_variables/case_log_variables.rb]: Is that a valid scenario?

**Comment by kosiakkatrina** [comment]
> Looks like there's still a looping query:
> 
> ```ruby
> Started POST "/logs/2026/form" for 127.0.0.1 at 2022-07-11 09:36:29 +0100
> Processing by FormController#submit_form as HTML
>   Parameters: {"authenticity_token"=>"[FILTERED]", "case_log"=>{"rent_type"=>"1", "irproduct_other"=>"", "page"=>"rent_type"}, "id"=>"2026"}
>   User Load (0.8ms)  SELECT "users".* FROM "users" WHERE "users"."id" = $1 ORDER BY "users"."id" ASC LIMIT $2  [["id", 2], ["LIMIT", 1]]
>   Organisation Load (0.7ms)  SELECT "organisations".* FROM "organisations" WHERE "organisations"."id" = $1 LIMIT $2  [["id", 1], ["LIMIT", 1]]
>   ↳ app/models/user.rb:44:in `case_logs'
>   CaseLog Load (0.9ms)  SELECT "case_logs".* FROM "case_logs" WHERE ("case_logs"."owning_organisation_id" = $1 OR "case_logs"."managing_organisation_id" = $2) AND "case_logs"."id" = $3 LIMIT $4  [["owning_organisation_id", 1], ["managing_organisation_id", 1], ["id", 2026], ["LIMIT", 1]]
>   ↳ app/controllers/form_controller.rb:96:in `find_resource'
>   TRANSACTION (1.3ms)  BEGIN
>   ↳ app/models/case_log.rb:518:in `reset_created_by'
>   User Load (0.9ms)  SELECT "users".* FROM "users" WHERE "users"."id" = $1 LIMIT $2  [["id", 2], ["LIMIT", 1]]
>   ↳ app/models/case_log.rb:518:in `reset_created_by'
>   CACHE Organisation Load (0.0ms)  SELECT "organisations".* FROM "organisations" WHERE "organisations"."id" = $1 LIMIT $2  [["id", 1], ["LIMIT", 1]]
>   ↳ app/models/case_log.rb:518:in `reset_created_by'
>   CACHE Organisation Load (0.0ms)  SELECT "organisations".* FROM "organisations" WHERE "organisations"."id" = $1 LIMIT $2  [["id", 1], ["LIMIT", 1]]
>   ↳ app/models/case_log.rb:520:in `reset_created_by'
>   Scheme Load (1.5ms)  SELECT "schemes".* FROM "schemes" WHERE "schemes"."id" = $1 LIMIT $2  [["id", 1], ["LIMIT", 1]]
>   ↳ app/models/derived_variables/case_log_variables.rb:9:in `scheme_has_multiple_locations?'
>   Location Count (1.4ms)  SELECT COUNT(*) FROM "locations" WHERE "locations"."scheme_id" = $1  [["scheme_id", 1]]
>   ↳ app/models/derived_variables/case_log_variables.rb:11:in `scheme_has_multiple_locations?'
>   CACHE Location Count (0.0ms)  SELECT COUNT(*) FROM "locations" WHERE "locations"."scheme_id" = $1  [["scheme_id", 1]]
>   ↳ app/models/derived_variables/case_log_variables.rb:11:in `scheme_has_multiple_locations?'
>   CACHE Location Count (0.1ms)  SELECT COUNT(*) FROM "locations" WHERE "locations"."scheme_id" = $1  [["scheme_id", 1]]
>   ↳ app/models/derived_variables/case_log_variables.rb:11:in `scheme_has_multiple_locations?'
>   CACHE Location Count (0.0ms)  SELECT COUNT(*) FROM "locations" WHERE "locations"."scheme_id" = $1  [["scheme_id", 1]]
> ```

@baarkerlounger do you mean `scheme_has_multiple_locations?` ? Cause it's way less than it was

**Comment by baarkerlounger** [comment]
@kosiakkatrina still looks to run ~400 times on submit for me locally?

File: app/models/derived_variables/case_log_variables.rb (hunk 1 above)
Comment (kosiakkatrina) [reply app/models/derived_variables/case_log_variables.rb]: I don't think so (it shouldn't be)

Comment (baarkerlounger) [reply app/models/derived_variables/case_log_variables.rb]: Should we change the tests to reflect that then?

**Pull request merged**